tests/haskell/executionTestLib/src/OpenCog/Lib.hs

```diff
@@ -20,5 +20,4 @@ foreign export ccall "somePredicate"
 c_pred = exportPredicate somePredicate
 
 somePredicate :: Atom -> TruthVal
-somePredicate (Link t xs tv) = tv
-somePredicate (Node t xs tv) = tv
+somePredicate (Link _ [Node _ _ tv] _) = tv
```

Comment:
@vsbogd, your fix seems odd. And I don't understand the syntax of square brackets in that context (I couldn't find a documentation on it). Could you elaborate as to why you think it is required?

Reply:
@ngeiswei , the issue I am trying to fix is that `somePredicate` in fact receives `ListLink` with parameters instead of single `Atom`. I believe previous version tries matching single argument in assumption that it is extracted from list and passed by `atomspace` implementation. But in fact `LibraryManager.cc` rebuilds `ListLink` of arguments keeping its content and passes it to the `somePredicate` without extracting internal atoms.

In other words I see two problems in here:
1. `SimpleTV` is set to `ListLink` instead of internal `ConceptNode` here:
https://github.com/singnet/atomspace/blob/33fcfb1557a6a62776e992e935baf8eb30cb7049/tests/haskell/haskellTest/Main.hs#L45-L49
2. `somePredicate` matches single `Atom` instead of `ListLink` containing atoms, here:
https://github.com/singnet/atomspace/blob/33fcfb1557a6a62776e992e935baf8eb30cb7049/tests/haskell/executionTestLib/src/OpenCog/Lib.hs#L22-L24

I expect that:
```hs
somePredicate (Link _ [Node _ _ tv] _) = tv
```
matches ListLink but looks like more accurate predicate definition is:
```hs
somePredicate (Link "ListLink" [(Node _ _ tv)] _) = tv
somePredicate (Link "ListLink" [(Link _ _ tv)] _) = tv
```
I expect `[]` match list of `ListLink` arguments.

Reply:
Indeed definition depends on `somePredicate` semantics and I assume that semantic is to return truth value of the only argument.

Reply:
I also considered a possibility of converting `ListLink` into native Haskell list but as this predicate is called as general `C` function I believe it is not an option.

Reply:
The unanswered question is why this code works on CircleCI. I am not sure because locally it always fails to me.

Reply:
It also fails for me.

I'm not familiar enough with the Haskell bindings (even though I mentored its initial implementation) to understand what is going on.

Reply:
I think the problem not in bindings but in the test predicate and test code. 
I see test is added by Alexey ( @Necr0X0der )  in https://github.com/singnet/atomspace/commit/43650bb791cebb64e08b5638869672d35be17672 commit and as far as I remember he was going to test only call of the predicate. So another way to fix test is to remove wrapping `ListLink` and pass just `ConceptNode`. I think I already tried it and it passes the test as well.

I see that this call https://github.com/singnet/atomspace/blob/master/opencog/atoms/execution/Force.cc made from `LibraryManager` should pass list unchanged or recreated depending on whether subatoms are executable. In test case atoms are not executable so list should return unchanged and test should pass (this is what is probably going on CircleCI). But locally it is recreated and we can look at the reason.

Reply:
Ok I think I need doing two things:
1. Ask @Necr0X0der whether it is ok to change the test: pass `ConceptNode` directly, or move `TruthValue` to `ConceptNode` and match it, or we are going to stick to the current test exactly
2. Find the reason why behavior is different on CirleCI and locally.

Reply:
Haskell binding was added just to test C++/lib backend for GroundedPredicate/Schema Nodes from other languages. I'm quite sure no one really uses it, so changing its API/tests is perfectly OK, if it is required by other more recent changes in Atomspace.

Reply:
Analysis for point (2) after deep dive into internals of `EvaluationLink` I found that in this case it always recreates `ListLink` wrapping `GroundedPredicateNode` (or any other predicate) arguments.
First `EvaluationLink` extracts `HandleSeq` from `ListLink` here (last line): https://github.com/singnet/atomspace/blob/33fcfb1557a6a62776e992e935baf8eb30cb7049/opencog/atoms/execution/EvaluationLink.cc#L820-L827
Then it wraps this `HandleSeq` into new `ListLink` here: https://github.com/singnet/atomspace/blob/33fcfb1557a6a62776e992e935baf8eb30cb7049/opencog/atoms/execution/EvaluationLink.cc#L771-L776
So there is no chance to keep `TruthValue` passed in the original test.

At the moment of the original test writing `EvaluationLink` code was different so test passed. `EvaluationLink` executed arguments here (line 301): https://github.com/singnet/atomspace/blob/43650bb791cebb64e08b5638869672d35be17672/opencog/atoms/execution/EvaluationLink.cc#L286-L304
And then passed it to the grounded function here (Line 708): https://github.com/singnet/atomspace/blob/43650bb791cebb64e08b5638869672d35be17672/opencog/atoms/execution/EvaluationLink.cc#L701-L715
So if argument is not executable there was a chance to keep original `TruthValue` in `ListLink`.

So now this test doesn't work because of change in atomspace API and I think we need to rewrite the test.

Reply:
Summary: there is no chance to pass `TruthValue` on `LinkedList` which wraps predicate arguments in current atomspace `EvaluationLink` implementation. So `somePredicate` definitely should match `ListLink` and `Node` or `Link` inside it. I don't see solution better than this PR.

Reply:
And here is CircleCI magic explanation:
```
test 165
 
    Start 165: HaskellAtomSpace
 

 
165: Test command: /bin/sh "runTest.sh" "/ws/atomspace/build/opencog/haskell" "/ws/atomspace/tests/haskell/haskellTest"
 
165: Test timeout computed to be: 10000000
 
165: Can't run Haskell-Tests as root
 
1/1 Test #165: HaskellAtomSpace .................   Passed    0.00 sec
 

 
The following tests passed:
 
	HaskellAtomSpace
 ```

Reply:
Looks like it should be fixed in separate PR